### DISCUSSION TITLE
Matrix multiplication explanation

### DISCUSSION THREAD

**Discussion by sudarshanmg** [melonjs/melonJS#1190] [Q&A] [answered]
Hey @obiot  If you don't mind, can you please explain how is the multiply method implemented in [`matrix2.js`](https://github.com/melonjs/melonJS/blob/master/src/math/matrix2.js#L157) file. I did not understand the process.

**Comment by obiot** [comment, marked as answer]
the matrix values are stored in a Float32Array (`val`).

The multiply method cache the reference to both arrays (`a`, and `b`,) and content of all values for both matrix (a0, a1,.... and b0, b1, ....), and then it's just to a typical multiplication. Final value are are set in the "original" `a` (this.val) matrix.
